### ISSUE TITLE
Python converters still not installed consistently

### ISSUE THREAD

**Issue by CoryMartin-NOAA** [JCSDA-internal/ioda-converters#1187]
## Description

Sorry to be beating the dead horse on this after the PR was merged that got tests passing...

After #1161 there is now the ability to `pip install` the ioda-converters, and scripts work with that approach. The install, however, instead of installing something broad like `pyiodaconv` takes all of the python subdirectories in `src/` and treats each as a python package. This means that there are installed libraries like so:
```
    /home/cmartin/.local/lib/python3.7/site-packages/compo/*
    /home/cmartin/.local/lib/python3.7/site-packages/conventional/*
    /home/cmartin/.local/lib/python3.7/site-packages/gnssro/*
    /home/cmartin/.local/lib/python3.7/site-packages/goes/*
    /home/cmartin/.local/lib/python3.7/site-packages/gsi_ncdiag/*
    /home/cmartin/.local/lib/python3.7/site-packages/hdf5/*
    /home/cmartin/.local/lib/python3.7/site-packages/ioda_converters-1.0.0.dist-info/*
    /home/cmartin/.local/lib/python3.7/site-packages/land/*
    /home/cmartin/.local/lib/python3.7/site-packages/lib_python/*
    /home/cmartin/.local/lib/python3.7/site-packages/marine/*
    /home/cmartin/.local/lib/python3.7/site-packages/ncep/*
    /home/cmartin/.local/lib/python3.7/site-packages/orig/*
    /home/cmartin/.local/lib/python3.7/site-packages/single_observation/*
    /home/cmartin/.local/lib/python3.7/site-packages/ssec/*
    /home/cmartin/.local/lib/python3.7/site-packages/wrfda_ncdiag/*
```
And that means things have to be imported like:
`import lib_python`

I find this concerning for a few reasons. 1) It's not clear what these libraries are, 2) something like 'hdf5' could conflict with a generic HDF5 library and 3) lib_python means nothing when these are all python libraries...

This is hacked together in the ctests by adding to `$PYTHONPATH` the `src/` directory because what used to work still places certain scripts into `build/lib/pyiodaconv`. These are inconsistent depending on how the converters are built/installed.

Instead, I would think this should be refactored such that there is just one installed library:
```
    /home/cmartin/.local/lib/python3.7/site-packages/pyiodaconv/*
```
and then the `import` statements should be consistent whether it is installed via pip or with ecbuild bundles.

`import pyiodaconv.lib_python`

## Requirements

IODA-converters should be installable and usable with both `pip` and `ecbuild` approach without having to add `src/` to `$PYTHONPATH`

## Acceptance Criteria (Definition of Done)
All of the following work without hacks
`pip install`
`ecbuild`

## Dependencies
None

**Comment by BenjaminRuston** [comment]
@CoryMartin-NOAA looks like this can be closed, please let me know if it needs to be reopened

**Comment by CoryMartin-NOAA** [comment]
@BenjaminRuston I put a hack in our GDASApp for now, but I don't think what was done in #1161 is a sufficient long term solution. It fixed the ctests and allowed us to proceed/improve but I don't think it's ideal. I'd like to get others who are python-savvy to chime in on this though. I just don't like 1) adding the `src/` directory to the `$PYTHONPATH` nor do I like `import lib_python` which is not a descriptive name.

**Comment by BenjaminRuston** [comment]
@CoryMartin-NOAA everyone wholeheartedly agrees with the `lib_python` name needs to be changed

and yes we can discuss, about how to approach the including `src/`, will have to ask other folks but would be nice to use the already existing method with `$PYTHONPATH` from the build directory so go back closer to what was there before.

**Comment by CoryMartin-NOAA** [comment]
Thanks @BenjaminRuston, this can be low priority for now, just wanted to open the issue to capture that this is something that needs to happen eventually.

**Comment by BenjaminRuston** [comment]
remind me and we can formulate a new issue Friday to capture this

**Comment by PatNichols** [comment]
@CoryMartin-NOAA @BenjaminRuston This will be addressed shortly in a new PR. I aiming for lib_python to be called pyiodaconv and installing in lib/python_version/pyioda/(name of module) Does that reasonable? Do you think the name is appropriate?

**Comment by CoryMartin-NOAA** [comment]
@PatNichols much better, yes! thanks for doing this

**Comment by CoryMartin-NOAA** [comment]
Tagging @danholdaway for his input/awareness to make sure this works for swell

**Comment by BenjaminRuston** [comment]
@CoryMartin-NOAA let's talk again about this think we may be able to use some of the other changes in  https://github.com/JCSDA-internal/ioda-converters/pull/1239 and https://github.com/JCSDA-internal/ioda/pull/1005

though the idea of the pip install is appealing think we may want to consolidate the code to be shared, and not create all the compo, conventional, ... directories

will try to find some time to discuss further but agree with your concerns and think we can come up with a solution and some steps toward it
